servicebox-app/src/main/java/io/halkyon/services/BindApplicationService.java

```diff
@@ -0,0 +1,75 @@
+package io.halkyon.services;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.halkyon.model.Application;
+import io.halkyon.model.Claim;
+import io.halkyon.model.Credential;
+import io.halkyon.model.CredentialParameter;
+import io.halkyon.model.Service;
+
+@ApplicationScoped
+public class BindApplicationService {
+
+    @Inject
+    KubernetesClientService kubernetesClientService;
+
+    public void bindApplication(Application application, Claim claim) {
+        Credential credential = getFirstCredentialFromService(claim.service);
+        String url = generateUrlByClaimService(application, claim);
+        if (credential != null && url != null) {
+            // scenario is supported
+            createSecretForApplication(application, claim, credential, url);
```

Comment:
Creating the secret is one thing but it is also needed that we change the Deployment resource to mount/inject the secret. Remark: Such a work is done by the SBO

Reply:
PR updated by mounting the secret in the deployment and injecting the data as env vars.